x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/scalar/nulls/CoalesceTests.java

```diff
@@ -44,18 +41,12 @@ public CoalesceTests(@Name("TestCase") Supplier<TestCase> testCaseSupplier) {
      */
     @ParametersFactory
     public static Iterable<Object[]> parameters() {
-        return parameterSuppliersFromTypedData(
-            new VaragsTestCases(
-                type -> "Coalesce",
-                strings -> Arrays.stream(strings)
-                    .filter(s -> s != null)
-                    .findFirst()
-                    .map(s -> equalTo((Object) new BytesRef(s)))
-                    .orElse(nullValue()),
-                longs -> Arrays.stream(longs).filter(s -> s != null).findFirst().map(l -> equalTo((Object) l)).orElse(nullValue()),
-                ints -> Arrays.stream(ints).filter(s -> s != null).findFirst().map(i -> equalTo((Object) i)).orElse(nullValue())
-            ).suppliers()
-        );
+        VaragsTestCaseBuilder builder = new VaragsTestCaseBuilder(type -> "Coalesce");
+        builder.expectString(strings -> strings.filter(v -> v != null).findFirst());
+        builder.expectLong(longs -> longs.filter(v -> v != null).findFirst());
+        builder.expectInt(ints -> ints.filter(v -> v != null).findFirst());
+        builder.expectBoolean(booleans -> booleans.filter(v -> v != null).findFirst());
```

Comment:
It isn't specific to the data type, but I think it could use all the cases too. I'd like to come at this in a follow up. I have some ideas on how to make sure we grab it.